### PR TITLE
MB-13718 convert tests to use transactions

### PR DESCRIPTION
**Pull request by mlm55**
## Summary

Converts the folowing testS to use transactions:
- ./pkg/payment_request
- ./pkg/route
- ./pkg/server
- ./pkg/parser/transittime
- ./pkg/services/ghcdieselfuelprice 
- ./pkg/services/fuelprice

[this article](https://transcom.github.io/mymove-docs/docs/backend/testing/running-server-tests-inside-a-transaction/) explains more about the approach used.

## Setup to Run Your Code

Run `make server_test` and make sure everything passes

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: pkg/services/mto_shipment/shipment_approver_test.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func (suite *MTOShipmentServiceSuite) TestApproveShipment() {
 		suite.Equal(models.MTOShipmentStatusApproved, fetchedShipment.Status)
 		suite.Equal(shipment.ID, fetchedShipment.ID)
 
-		err = appCtx.DB().EagerPreload("ReService").Where("mto_shipment_id = ?", shipmentForAutoApprove.ID).All(&serviceItems)
+		err = appCtx.DB().EagerPreload("ReService").Where("mto_shipment_id = ?", shipmentForAutoApprove.ID).Order("created_at asc").All(&serviceItems)
```
Comment (mlm55) [inline pkg/services/mto_shipment/shipment_approver_test.go:136]: This is an attempt to fix a [flaky test](https://app.circleci.com/pipelines/github/transcom/mymove/44932/workflows/375a93fa-6989-4ddc-9b6e-444f53977c19/jobs/712368/tests#failed-test-0)

**Comment by shimonacarvalho** [comment]
I was actually unable to get the tests to pass. I thought it was my setup at first but I managed to run main and pass everything. Can you check that the tests pass? Errors were largely in `TestAllGenerateEdi`

**Review by shimonacarvalho — APPROVED** [review]
Looks good, just wondered why the filename change?

File: pkg/parser/transittime/parse_transit_times_test.go [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-openapi/swag"
+	"go.uber.org/zap"
+)
+
+func (suite *TransitTimeParserSuite) Test_xlsxDataSheetInfo_generateOutputFilename() {
+
```
[the remark this answers is not in the document]
Comment (mlm55) [reply pkg/parser/transittime/parse_transit_times_test.go:12]: This was originally in `pkg/parser/transittime/parse_test.go` and I had moved it to a new file (to match the pattern where the test suite setup was in its own file) and named it incorrectly. But, looking at it more closely, some of these tests test `parse.go` and some test `shared.go`. I'll separate them out and rename the file.

**Pull request merged**